PasswordRules.cs

```diff
@@ -0,0 +1,151 @@
+            {
+                if ((Char.IsLetter(c)) && (Char.IsUpper(c)))
+                    return true;
+            }
+            Console.WriteLine("Password does not contain at least one Uppercase letter");
+            return false;
+        }
+
+        private bool ContainsLowerLetter(string pass)
+        {
+            foreach (char c in pass)
+            {
+                if ((Char.IsLetter(c)) && (Char.IsLower(c)))
+                    return true;
+            }
+            Console.WriteLine("Password does not contain at least one Lowercase letter");
+            return false;
+        }
+
+        private bool ContainsDigit(string pass)
+        {
+            foreach (char c in pass)
+            {
+                if (Char.IsDigit(c))
+                    return true;
+            }
+            Console.WriteLine("Password does not contain at least one digit");
+            return false;
+        }
+
+        private bool ContainsSymbol(string pass) // 
+        {
+            string validPattern = ".!@#$%^&*";
+            foreach (char c in pass)
+            {
+                if (validPattern.Contains(c))
+                    return true;
+            }
+            Console.WriteLine("Password does not contain a special symbol");
+            return false;
+        }
+
+        private bool ContainsTenCharacters(string pass)
+        {
+            if (pass.Length >= 10)
+            {
+                return true;
+            }
+            else
+            {
+                Console.WriteLine("Password does not contain at least 10 characters");
+                return false;
+            }
+        }
+
+
+        public bool CallTheAllCheckFunctions(string pass)
+        {
+            if (!ContainsTenCharacters(pass))
+            {
+                return false;
+            }
+            if (!ContainsSymbol(pass))
+            {
+                return false;
+            }
+            if (!ContainsDigit(pass))
+            {
+                return false;
+            }
+            if (!ContainsLowerLetter(pass))
+            {
+                return false;
+            }
+            if (!ContainsUpperLetter(pass))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public void CheckPasswordByApi(string pass)
+        {
+            string sHash = string.Empty;
+
+            using (SHA1Managed sha1 = new SHA1Managed()) // use the class to hash the password 
+            {
+                var hash = sha1.ComputeHash(Encoding.UTF8.GetBytes(pass)); // create the hash as a bit array
+                sHash = BitConverter.ToString(hash).Replace("-", ""); // convert bit array to hexadecimal
+            }
+
+            string existingPasswords = requestThroughApi(sHash.Substring(0, 5)); //call the function for API request
+            if (!string.IsNullOrWhiteSpace(existingPasswords)) // if string returned, we will check our password

```

Comment:
I guess this is pretty redundant, according to HaveIBeenPwned documentation:

> On average, a range search returns 478 hash suffixes, although this number will differ depending on the hash prefix being searched for. The smallest result is 381, the largest 584. There are 1,048,576 different hash prefixes between 00000 and FFFFF (16^5) and every single one will return HTTP 200; there is no circumstance in which the API should return HTTP 404.

Which means that there will be at least 381 results in response and it should not ever return null/empty string/whitespace.
We could rather check if `HttpStatusCode `of the response is `200` :)